internal/io/edgex/edgex_sink_test.go

```diff
@@ -598,8 +598,9 @@ func TestEdgeXTemplate_Apply(t1 *testing.T) {
 		var payload []map[string]interface{}
 		json.Unmarshal([]byte(t.input), &payload)
 		dt := t.conf["dataTemplate"]
-		tf, _ := transform.GenTransform(cast.ToStringAlways(dt), "json", "", "")
+		tf, sf, _ := transform.GenTransform(cast.ToStringAlways(dt), "json", "", "", []string{})
```

Comment:
In normal cases (like MQTT), the 3 properties are run in order dataTemplate -> fields -> format. Currently dataTemplate/format are handled in transformFunc. I think fields is also part of the transform. Any benefits to mvoe it out? The sf may not work because tf will return bytes

Reply:
I previously thought that plugin sink could override this select function, but it seems not necessary at present. I will modify the code according to your suggestion.